google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminClient.java

```diff
@@ -1463,9 +1434,226 @@ public Policy apply(com.google.iam.v1.Policy proto) {
+   *     IAM management</a>
+   */
+  @SuppressWarnings("WeakerAccess")
+  public ApiFuture<Policy> setBackupIamPolicyAsync(
+      String clusterId, String backupId, Policy policy) {
+    String backupName = NameUtil.formatBackupName(projectId, instanceId, clusterId, backupId);
+    return setIamPolicy(policy, backupName);
+  }
+
+  /**
+   * Tests whether the caller has the given permissions for the specified backup. Returns a subset
+   * of the specified permissions that the caller has.
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * List<String> grantedPermissions = client.testBackupIamPermission("my-cluster-id", "my-backup-id",
+   *   "bigtable.backups.restore", "bigtable.backups.delete");
+   * }</pre>
+   *
+   * System.out.println("Has restore access: " +
+   * grantedPermissions.contains("bigtable.backups.restore"));
+   *
+   * <p>System.out.println("Has delete access: " +
+   * grantedPermissions.contains("bigtable.backups.delete"));
+   *
+   * @see <a href="https://cloud.google.com/bigtable/docs/access-control#permissions">Cloud Bigtable
+   *     permissions</a>
+   */
+  @SuppressWarnings({"WeakerAccess"})
+  public List<String> testBackupIamPermission(
+      String clusterId, String backupId, String... permissions) {
+    return ApiExceptions.callAndTranslateApiException(
+        testBackupIamPermissionAsync(clusterId, backupId, permissions));
+  }
+
+  /**
+   * Asynchronously tests whether the caller has the given permissions for the specified backup.
+   * Returns a subset of the specified permissions that the caller has.
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * ApiFuture<List<String>> grantedPermissionsFuture = client.testBackupIamPermissionAsync("my-cluster-id", "my-backup-id",
+   *   "bigtable.backups.restore", "bigtable.backups.delete");
+   *
+   * ApiFutures.addCallback(grantedPermissionsFuture,
+   *   new ApiFutureCallback<List<String>>() {
+   *     public void onSuccess(List<String> grantedPermissions) {
+   *       System.out.println("Has restore access: " + grantedPermissions.contains("bigtable.backups.restore"));
+   *       System.out.println("Has delete access: " + grantedPermissions.contains("bigtable.backups.delete"));
+   *     }
+   *
+   *     public void onFailure(Throwable t) {
+   *       t.printStackTrace();
+   *     }
+   *   },
+   *   MoreExecutors.directExecutor());
+   * }</pre>
+   *
+   * @see <a href="https://cloud.google.com/bigtable/docs/access-control#permissions">Cloud Bigtable
+   *     permissions</a>
+   */
+  @SuppressWarnings({"WeakerAccess"})
+  public ApiFuture<List<String>> testBackupIamPermissionAsync(
+      String clusterId, String backupId, String... permissions) {
+    String backupName = NameUtil.formatBackupName(projectId, instanceId, clusterId, backupId);
+    return testIamPermissions(backupName, permissions);
+  }
+
+  private ApiFuture<Policy> getIamPolicyResult(String name) {
```

Comment:
The name sticks out a bit. I'm guessing its to avoid conflict with existing table getIamPolicy.

I think it would be cleaner to name the generic versions as `getResourceIamPolicy`, `setResourceIamPolicy`, etc. or `getIamPolicyBase`

And the public resource specific ones can be called `getBackupIamPolicy`.